pkg/apiserver/vshn/redis/get.go

```diff
@@ -0,0 +1,54 @@
+package redis
+
+import (
+	"context"
+	"fmt"
+
+	appcatv1 "github.com/vshn/appcat/apis/appcat/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apiserver/pkg/endpoints/request"
+	"k8s.io/apiserver/pkg/registry/rest"
+)
+
+var _ rest.Getter = &vshnRedisBackupStorage{}
+
+func (v *vshnRedisBackupStorage) Get(ctx context.Context, name string, opts *metav1.GetOptions) (runtime.Object, error) {
+
+	namespace, ok := request.NamespaceFrom(ctx)
+	if !ok {
+		return nil, fmt.Errorf("cannot get namespace from context")
+	}
+
+	instances, err := v.vshnRedis.ListVSHNRedis(ctx, namespace)
+	if err != nil {
+		return nil, err
+	}
+
+	redisSnap := &appcatv1.VSHNRedisBackup{}
+
+	for _, instance := range instances.Items {
+		ins := instance.Status.InstanceNamespace
+		snap, err := v.snapshothandler.Get(ctx, name, ins)
+		if err != nil {
+			if apierrors.IsNotFound(err) {
+				continue
+			}
+			return nil, err
+		}
+
+		redisSnap = &appcatv1.VSHNRedisBackup{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: trimStringLenght(*snap.Spec.ID),
+			},
```

Comment:
I see two issues with this:

* Why do you take the beginning of the ID instead of the name of the backup? They are the same right? If they aren't things will be weird. I.e. you list backups and then try to get one based on the name but you won't be able to as the name is different
* You should set more then just the name in the object meta. At least the namespace should be set. But generally we expect `uid`, `resourceVersion`, `generation`, and `creationTimestamp` to be set. Some tooling might expect this to be present and things might fail in very unexpected ways. I'd propose to just copy the complete metadata from the snapshot and update the namespace

Also more of a k8up issue: How likely is it that two snapshots have the same name? That could result in an issue...

Reply:
> Also more of a k8up issue: How likely is it that two snapshots have the same name? That could result in an issue...

This should really not happen at all, as a clash in the snapshot IDs would completely break a restic repository.

Reply:
> You should set more then just the name in the object meta. At least the namespace should be set. But generally we expect uid, resourceVersion, generation, and creationTimestamp to be set. Some tooling might expect this to be present and things might fail in very unexpected ways. I'd propose to just copy the complete metadata from the snapshot and update the namespace

Shoulddn't I create a new UID in this case though? This feels like it could be problematic as well...

Reply:
> Shoulddn't I create a new UID in this case though? This feels like it could be problematic as well...

I remember we had a longer discussion about this for the initial control API and found after longer digging that UIDs only need to be unique per Kind. So reusing the UID from the snapshots solves our problem of unique IDs without having to generate new once and having to track more state.

Also OpenShift does the exact same thing for `Projects` and that seemed to have worked fine until now :)

Reply:
> This should really not happen at all, as a clash in the snapshot IDs would completely break a restic repository.

But as far as I can tell we're only using the first few characters of the snapshot ID as the name. So there could be clashes. Not likely so not a huge issue but still

Reply:
> > This should really not happen at all, as a clash in the snapshot IDs would completely break a restic repository.
> 
> But as far as I can tell we're only using the first few characters of the snapshot ID as the name. So there could be clashes. Not likely so not a huge issue but still

True, but printing the whole ID is waaay too long and we currently don't have another field we could use as a name. So for now let's stick with this.

Reply:
> > Shoulddn't I create a new UID in this case though? This feels like it could be problematic as well...
> 
> I remember we had a longer discussion about this for the initial control API and found after longer digging that UIDs only need to be unique per Kind. So reusing the UID from the snapshots solves our problem of unique IDs without having to generate new once and having to track more state.
> 
> Also OpenShift does the exact same thing for `Projects` and that seemed to have worked fine until now :)

Yeah makes it easier. I did a some test with generating a stable UID based on the name/id of the snapshot, but if re-using the UID is no issue I'll just do that.